keps/sig-network/4962-network-topology-standard/README.md

```diff
@@ -0,0 +1,938 @@
+
+Cluster topology information can be derived from various sources:
+- Provided directly by a Cloud Service Provider (CSP)
+- Extracted from a CSP using specialized tools like [topograph](https://github.com/NVIDIA/topograph)
+- Manually set up by cluster administrators
+- A combination of the above methods to ensure comprehensive coverage
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Reserved Network Types
+We have introduced reserved network types to better accommodate common network hierarchies. These reserved network types include the following predefined names and characteristics:
+
+1. `accelerator`: Network interconnect for direct accelerator communication (e.g., Multi-node NVLink interconnect between NVIDIA GPUs)
+2. `block`: Rack-level switches connecting hosts in one or more racks as a block.
+3. `datacenter`: Spine-level switches connecting multiple blocks inside a datacenter.
+4. `zone`: Zonal switches connecting multiple datacenters inside an availability zone.
+
+When using reserved network types, Network QoS Annotations become optional. In the absence of these annotations, it is assumed that performance within each network layer is uniform.
+
+The scheduler will prioritize switches according to the order outlined above, providing a standardized approach for network-aware scheduling across a range of configurations.
+
+If provided, Network QoS Annotations can be used to refine and enhance the details of link performance, enabling more precise scheduling decisions.
+
+#### Example 1: network topology representation with reserved network types:
+
+Consider the following network topology:
+
+![Netwotk topology with reserved network types](./img/topo-reserved-labels.png)
+
+Let's examine node `vm12` as an example. This node is connected to NVSwitch `nvl10` and network switch `sw11`, which in turn is connected to switches `sw21` and `sw31`.
+In this case, the node labels would be:
+```yaml
+network.topology.kubernetes.io/accelerator: nvl10
+network.topology.kubernetes.io/block: sw11
+network.topology.kubernetes.io/datacenter: sw21
+network.topology.kubernetes.io/zone: sw31
+```
+
+If we have additional information such as latency and/or bandwidth between the node and the switches, it can be provided in an annotation:
+```yaml
+network.qos.kubernetes.io/switches: {
+   "nvl10": {
+      "latency": "2us",
+      "bandwidth": "100Gbps"
+   },
+   "sw11": {
+      "latency": "50us",
+      "bandwidth": "40Gbps"
+   },
+   "sw21": {
+      "latency": "500us",
+      "bandwidth": "20Gbps"
+   },
+  "sw31": {
+      "latency": "1ms",
+      "bandwidth": "10Gbps"
+   }
```

Comment:
These are network interfaces on the node, I think we should better model them with DRA https://github.com/kubernetes/enhancements/pull/4965/files#r1846865095 , that also allows us to provide dynamic capabilities to the interfaces

Reply:
Again, as mentioned in the earlier comment, these QoS numbers represent node-to-reachable-switch metrics. They are not per NIC.

Reply:
does it mean that some switches may not be connected directly?
how then the latency and bandwidth are obtained to guarantee those values?